tests/sentry/models/test_organization.py

```diff
@@ -420,6 +468,43 @@ def test_absolute_url_with_customer_domain(self):
         assert url == "http://acme.testserver/issues/?project=123#ref"
 
 
+@region_silo_test(stable=True)
+class OrganizationUpdateOverrideTest(TestCase):
+    def test_slug_change_on_org_update_in_monolith_mode(self):
+        if SiloMode.get_current_mode() != SiloMode.MONOLITH:
+            return
+
+        org = self.create_organization(name="santry", slug="santry_slug")
+        assert org.slug == "santry-slug"
+        update_params = dict(slug="new santry")
+        org.update(**update_params)
+        assert org.slug == "new-santry"
+        org.refresh_from_db()
+        assert org.slug == "new-santry"
+
+    def test_slug_change_on_org_update_in_region_silo(self):
+        if SiloMode.get_current_mode() != SiloMode.REGION:
+            return
+
+        org = Organization.objects.create(name="santry", slug="santry_slug")
+        assert org.slug == "r-na-santry-slug"
+        update_params = dict(slug="new santry")
+        org.update(**update_params)
+        assert org.slug == "r-na-new-santry"
+        org.refresh_from_db()
+        assert org.slug == "r-na-new-santry"
```

Comment:
When we're running as split silos how do global (unprefixed) slugs get assigned? This test shows an update to an existing record that results in a r-na prefix being assigned.

Reply:
That's a good question. One option is a new `set_vanity_slug` method on org that effectively bypasses this check, but I want to find a way to guard against this being called in normal code as it should really only be invoked via a control provisioning flow 🤔 

Any other ideas are welcome though!

Reply:
With the region slug prefixing logic being integrated into save() and update() how would additional methods by-pass the logic in those methods though?